x/sysproxy/sysproxy_windows.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2024 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build windows
+
+package sysproxy
+
+import (
+	"errors"
+	"fmt"
+
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/registry"
+)
+
+var (
+	modwininet            = windows.NewLazySystemDLL("wininet.dll")
+	procInternetSetOption = modwininet.NewProc("InternetSetOptionW")
+)
+
+// https://learn.microsoft.com/en-us/windows/win32/wininet/option-flags
+// INTERNET_OPTION_SETTINGS_CHANGED: 39
+// Notifies the system that the registry settings have been changed so that it verifies the settings on the next call to InternetConnect.
+// INTERNET_OPTION_REFRESH: 37
+// Causes the proxy data to be reread from the registry for a handle. No buffer is required.
+// This option can be used on the HINTERNET handle returned by InternetOpen.
+// This is used by InternetSetOption.
+const (
+	INTERNET_OPTION_SETTINGS_CHANGED = 39
+	INTERNET_OPTION_REFRESH          = 37
+)
+
+// https://learn.microsoft.com/en-us/windows/win32/api/wininet/nf-wininet-internetsetoptionw
+// internetSetOption sets an Internet option.
+func internetSetOption(hInternet uintptr, dwOption int, lpBuffer uintptr, dwBufferLength uint32) bool {
+	ret, _, _ := procInternetSetOption.Call(
+		hInternet,
+		uintptr(dwOption),
+		lpBuffer,
+		uintptr(dwBufferLength),
+	)
+	return ret != 0
+}
+
+func resetWininetProxySettings() error {
+
+	result1 := internetSetOption(0, INTERNET_OPTION_SETTINGS_CHANGED, 0, 0)
```

Comment:
Change to error, and return early if failed.

```suggestion
	if err := internetSetOption(0, INTERNET_OPTION_SETTINGS_CHANGED, 0, 0); err != nil {
		return fmt.Errorf("call to InternetSetOptionW(INTERNET_OPTION_SETTINGS_CHANGED, 0, 0) failed: %w", err)
	}
```